evaluate/task/task-code-repair.go

```diff
@@ -0,0 +1,158 @@
+// This task requires the repository to consist of multiple packages, with the code repair logic applied to each package.
+func (t *TaskCodeRepair) Run(repository evaltask.Repository) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+	log, logClose, err := log.WithFile(t.Logger, filepath.Join(t.ResultPath, string(t.Identifier()), model.CleanModelNameForFileSystem(t.Model.ID()), t.Language.ID(), repository.Name()+".log"))
+	if err != nil {
+		return nil, nil, err
+	}
+	defer logClose()
+
+	log.Printf("Evaluating model %q on task %q using language %q and repository %q", t.Model.ID(), t.Identifier(), t.Language.ID(), repository.Name())
+	defer func() {
+		log.Printf("Evaluated model %q on task %q using language %q and repository %q: encountered %d problems: %+v", t.Model.ID(), t.Identifier(), t.Language.ID(), repository.Name(), len(problems), problems)
+	}()
+
+	var packagePaths []string
+	files, err := os.ReadDir(repository.DataPath())
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	for _, file := range files {
+		if file.IsDir() && !strings.HasPrefix(file.Name(), ".") {
+			packagePaths = append(packagePaths, filepath.Join(repository.DataPath(), file.Name()))
+		}
+	}
+
+	repositoryAssessment = metrics.NewAssessments()
+	for _, packagePath := range packagePaths {
+		if err := repository.Reset(t.Logger); err != nil {
+			t.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
+		}
+
+		sourceFile, mistakes, err := t.validateCodeRepairCase(packagePath, repository, log)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ctx := evaltask.Context{
+			Language: t.Language,
+
+			RepositoryPath: packagePath,
+			FilePath:       sourceFile,
+
+			Arguments: &TaskArgumentsCodeRepair{
+				Mistakes: mistakes,
+			},
+
+			Logger: log,
+		}
+		assessments, err := t.Model.RunTask(ctx, t.Identifier())
+		if err != nil {
+			problems = append(problems, pkgerrors.WithMessage(err, sourceFile))
+
+			continue
+		}
+		if assessments[metrics.AssessmentKeyProcessingTime] == 0 {
+			return nil, nil, pkgerrors.Errorf("no model response time measurement present for %q at repository %q", t.Model.ID(), repository.Name())
+		}
+		repositoryAssessment.Add(assessments)
+		repositoryAssessment.Award(metrics.AssessmentKeyResponseNoError)
+
+		coverage, ps, err := t.Language.Execute(log, packagePath)
+		problems = append(problems, ps...)
+		if err != nil {
+			problems = append(problems, pkgerrors.WithMessage(err, sourceFile))
+
+			continue
+		}
+		log.Printf("Executes tests with %d coverage objects", coverage)
+		repositoryAssessment.Award(metrics.AssessmentKeyFilesExecuted)
+		repositoryAssessment.AwardPoints(metrics.AssessmentKeyCoverage, coverage)
+	}
+
+	return repositoryAssessment, problems, nil
+}
+
+func (t *TaskCodeRepair) validateCodeRepairCase(packagePath string, repository evaltask.Repository, fileLogger *log.Logger) (sourceFile string, mistakes []string, err error) {
```

Comment:
I think it is a good idea to already extract the source file path and mistakes within here. But I think we need a better name for the function, then :thinking:. Cause it is not only validating but also like... "unpacking" the test case... or "extracting"? Please pick what sounds right (maybe you come up with something even better) and also add a docstring.